src/webgl/p5.Framebuffer.js

```diff
@@ -0,0 +1,552 @@
+import p5 from '../core/main';
+import * as constants from '../core/constants';
+import { checkWebGLCapabilities } from './p5.Texture';
+
+class FramebufferCamera extends p5.Camera {
+  constructor(fbo) {
+    super(fbo.target._renderer);
+    this.fbo = fbo;
+  }
+
+  _computeCameraDefaultSettings() {
+    super._computeCameraDefaultSettings();
+    this.defaultAspectRatio = this.fbo.width / this.fbo.height;
+    this.defaultEyeZ =
+      this.fbo.height / 2.0 / Math.tan(this.defaultCameraFOV / 2.0);
+    this.defaultCameraNear = this.defaultEyeZ * 0.1;
+    this.defaultCameraFar = this.defaultEyeZ * 10;
+  }
+
+  resize() {
+    // If we're using the default camera, update the aspect ratio
+    if (this.cameraType === 'default') {
+      this._computeCameraDefaultSettings();
+      this._setDefaultCamera();
+    } else {
+      this.perspective(
+        this.cameraFOV,
+        this.fbo.width / this.fbo.height
+      );
+    }
+  }
+}
+
+p5.FramebufferCamera = FramebufferCamera;
+
+class FramebufferTexture {
+  constructor(framebuffer, property) {
+    this.framebuffer = framebuffer;
+    this.property = property;
+  }
+
+  get width() {
+    return this.framebuffer.width * this.framebuffer.density;
+  }
+
+  get height() {
+    return this.framebuffer.height * this.framebuffer.density;
+  }
+
+  rawTexture() {
+    return this.framebuffer[this.property];
+  }
+}
+
+p5.FramebufferTexture = FramebufferTexture;
+
+class Framebuffer {
+  /**
+   * @class p5.Framebuffer
+   * @constructor
+   * @param {p5.Graphics|p5} target A p5 global instance or p5.Graphics
+   * @param {Object} [settings] A settings object
+   */
+  constructor(target, settings = {}) {
+    this.target = target;
+    this.target._renderer.framebuffers.add(this);
+
+    this.format = settings.format || constants.UNSIGNED_BYTE;
+    this.channels = settings.channels || (
+      target._renderer._pInst._glAttributes.alpha
+        ? constants.RGBA
+        : constants.RGB
+    );
+    this.depthFormat = settings.depthFormat || (
+      this.format === constants.UNSIGNED_BYTE
```

Comment:
Same here, we should definitely include half_float for the base format.

Reply:
Assuming there's only one float format for depth (f32), the behaviour right now is: depth tries to match as close as possible the color format, so if color is UNSIGNED_BYTE, then depth will also be UNSIGNED_BYTE, and if color is FLOAT or HALF_FLOAT then depth will also be FLOAT.

Reply:
got it, I think in the absence of hal_float as a depth format you can disregard my comments here